### DISCUSSION TITLE
publishing a site on github

### DISCUSSION THREAD

**Discussion by punjoe** [community/community#29439] [Other Feature Feedback, Questions, & Ideas] [answered]
hi can someone please help, I am new to github, and trying to publish a site i have made with html and css, however when i go to the link of my published page it only shows my page in html and no css styling, even though i have added all the files on the repository, my index.html file my css.styles file and all my images. can someone please help

**Comment by airtower-luna** [comment]
If you post the link to the repository maybe someone will have an idea. Usually it's a broken URL somewhere. :wink:

**Reply by punjoe** [reply]
https://github.com/punjoe/puns-developers-site is this what you mean??

**Reply by punjoe** [reply]
https://punjoe.github.io/puns-developers-site/ this is the url as you can see it is only showing html

**Reply by airtower-luna** [reply]
Having both of those is always best. The first is your repository, the second the deployed site. :slightly_smiling_face: 

Your problem is the `href` here: https://github.com/punjoe/puns-developers-site/blob/8136f41fc8bb0c6a5e1fe7b505bee3be66a641e7/index.html#L9

```html
<link rel="stylesheet" href="github.com/punjoe/puns-developers-site/blob/main/styles.css">
```

It tells the browser to look for the `styles.css` file in a folder `github.com/punjoe/[...]/main/` *relative to* the folder containing `index.html`.

The best thing you can do here is use a correct relative link. You have `styles.css` in the same folder as `index.html`, so it's simply:

```html
<link rel="stylesheet" href="styles.css">
```

In theory, if you want to link to a random website, you need to include the scheme, e.g. `https://`.

**Reply by punjoe** [reply]
i have tried this and it is still not working I've changed the link to the stylesheet exactly how you said within the html changing from <link rel="stylesheet" href="github.com/punjoe/puns-developers-site/blob/main/styles.css"> to 
<link rel="stylesheet" href="styles.css">

**Reply by airtower-luna** [reply]
I can see the CSS is loading fine now. :slightly_smiling_face: 

There are definitely additional (different) issues. In particular, pretty much all your `<img>` tags point at files on your disk, not in your repository. For example here: https://github.com/punjoe/puns-developers-site/blob/d65e9ee2de5b9eec0cb358e0af9baf6c0de998d0/index.html#L19

```html
<img class="top-cloud" src="C:\Users\44793\OneDrive\Desktop\web development\CSS-Pun's developer site\images\cloud.png" alt="cloud-img">
```

A random person opening your site can't access your `C:` drive. You need to add the images to your repository and use relative links to them. Keep in mind that internet URLs are case sensitive (so `Images/` and `images/` are different) and must use forward slashes (`/`) as the directory separator.

**Reply by airtower-luna** [reply]
Looks like the images are working now, only the favicon still has `href="C:\..."`. :slightly_smiling_face:

**Comment by delosemre** [comment, marked as answer]
Hi, you need to write the addresses of the pictures correctly. 

`          <img class="skillsonepic" src="C:\Users\44793\OneDrive\Pictures\Screenshots\webdevloper.png" alt="picture-of -developer">
`
This is wrong.
`<img class="skillsonepic" src="/images/webdevloper.png" alt="picture-of -developer">`
Indeed it is like this.

I hope I have helped. 🙂

**Reply by airtower-luna** [reply]
That absolute path (starting with `/`) actually wouldn't work. The site as a whole is deployed under `/puns-developers-site/`, so all absolute paths would have to start with that. As a rule of thumb, avoid absolute paths and use relative paths as far as reasonably possible.

**Reply by punjoe** [reply]
thanks for all your help everyone its working now

**Reply by PavelLaptev** [reply]
@punjoe can you please mark the question as answered if the issue is solved :-)
